From 1b86627679a39c05a6abb63dd86c2fa712560397 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Thu, 27 Sep 2018 23:33:27 +0200
Subject: [PATCH 5/7] 
 0001-Move-cache-cleanup-out-into-a-common-function-and-cl.patch

---
 src/daemon.c | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/src/daemon.c b/src/daemon.c
index ae59e50..19ad922 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -140,6 +140,20 @@ error_get_type (void)
 #define MAX_LOCAL_USERS 50
 #endif
 
+static void
+remove_cache_files (const gchar *user_name)
+{
+        gchar *filename;
+
+        filename = g_build_filename (USERDIR, user_name, NULL);
+        g_remove (filename);
+        g_free (filename);
+
+        filename = g_build_filename (ICONDIR, user_name, NULL);
+        g_remove (filename);
+        g_free (filename);
+}
+
 static struct passwd *
 entry_generator_fgetpwent (Daemon       *daemon,
                            GHashTable   *users,
@@ -1155,7 +1169,6 @@ daemon_uncache_user_authorized_cb (Daemon                *daemon,
                                    gpointer               data)
 {
         const gchar *user_name = data;
-        gchar       *filename;
         User        *user;
 
         sys_log (context, "uncache user '%s'", user_name);
@@ -1170,13 +1183,7 @@ daemon_uncache_user_authorized_cb (Daemon                *daemon,
         /* Always use the canonical user name looked up */
         user_name = user_get_user_name (user);
 
-        filename = g_build_filename (USERDIR, user_name, NULL);
-        g_remove (filename);
-        g_free (filename);
-
-        filename = g_build_filename (ICONDIR, user_name, NULL);
-        g_remove (filename);
-        g_free (filename);
+        remove_cache_files (user_name);
 
         accounts_accounts_complete_uncache_user (NULL, context);
 
@@ -1216,7 +1223,6 @@ daemon_delete_user_authorized_cb (Daemon                *daemon,
 {
         DeleteUserData *ud = data;
         GError *error;
-        gchar *filename;
         struct passwd *pwent;
         const gchar *argv[6];
 
@@ -1243,13 +1249,7 @@ daemon_delete_user_authorized_cb (Daemon                *daemon,
 
         }
 
-        filename = g_build_filename (USERDIR, pwent->pw_name, NULL);
-        g_remove (filename);
-        g_free (filename);
-
-        filename = g_build_filename (ICONDIR, pwent->pw_name, NULL);
-        g_remove (filename);
-        g_free (filename);
+        remove_cache_files (pwent->pw_name);
 
         argv[0] = "/usr/sbin/userdel";
         if (ud->remove_files) {
-- 
2.19.0

